### DISCUSSION TITLE
Maintenance mode?

### DISCUSSION THREAD

**Discussion by hems** [vercel/vercel#8744] [Ideas]
Heroku has this really handy Maintenance mode feature which allows you to fallback to a static file in case you need to do heavy updates or temporarily put the website down.

I believe it would be great to have such feature on vercel otherwise users have to implement it manually which would require a solution such as having an environment variable for Maintenance and then reading it in real time from the Vercel API and then making sure your app is prepared to deal with such update.

Any thoughts?

**Comment by amyegan** [comment]
Great suggestion! You should be able to use a Project config file to [rewrite](https://vercel.com/docs/project-configuration#project-configuration/rewrites) or [redirect](https://vercel.com/docs/project-configuration#project-configuration/redirects) to a maintenance page, but I can see how it would be nice to have a UI toggle. I'm adding this to the requested features list 😃

**Reply by hems** [reply]
> Great suggestion! You should be able to use a Project config file to [rewrite](https://vercel.com/docs/project-configuration#project-configuration/rewrites) or [redirect](https://vercel.com/docs/project-configuration#project-configuration/redirects) to a maintenance page, but I can see how it would be nice to have a UI toggle. I'm adding this to the requested features list 😃

that's a lovely suggestion to use the project config, really good solution, still would require an entire deploy which is fine most of the times but not fine in emergencies for instance.

thank you
